### PR TITLE
Android: Let the OS handle the hardware buttons

### LINKED ISSUE #2122: Volume key Android
On Android port Volume key don't work.
Tested on more device.
Don't change volume and don't show volume panel,
Pls, fix.
And Volume (+ -) and power (screenshot on more ROM) = error on chat!

### PR DESCRIPTION
**Pull request by est31**
Should fix the same bugs as  #4086, but that PR has some issues. I'd have waited for @MoNTE48 to fix them but I rather want the release to be done (without rushing of course).

Would be cool if anyone can confirm that this fixes #2122 and #1454. @Zeno- ? @Wayward1?

### CODE REVIEW

**Comment by est31** [comment]
Adding own :+1:

**Comment by sofar** [comment]
So I (somehow, by magic) made a working minetest debug APK build and loaded it on my sons kindle fire with this pach applied.

This model fire has a hardware volume rocker.

Using it does nothing. The volume isn't adjusted. I can not take screenshots.

Pushing the back button shows me the pause window.

again, this is ec15e35ad7effe54d04c0d34ead3bc63d0e311f1 + 4119 applied

**Comment by Wayward1** [comment]
This seems to work well, except pressing the back button bypasses the pause menu and exits the app. The pause menu is shown for a split second, then the app closes.

**Comment by MoNTE48** [comment]
Why is not anyone listening to me?

**Comment by paramat** [comment]
We're listening, any input i sappreciated.

**Comment by MoNTE48** [comment]
I said why do we need onbackpresed, but est31 decided that he knows more. So I'm sure he did not tested before posting your pr

**Comment by DonBatman** [comment]
@Monte48 est31 is doing the code and myself and others are testing.

**Comment by paramat** [comment]
See IRC minetest-project channel for testing discussion http://minetest.org/ircproject/2016-05-13.html

**Comment by paramat** [comment]
@est31 #4086 has been updated, please see the latest commit.

File: build/android/patches/irrlicht-back_button.patch [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-@@ -423,6 +423,7 @@
+@@ -486,7 +488,7 @@
+ 				event.KeyInput.Char = 0;
  			}
- 			
- 			device->postEventFromUser(event);
-+			status = 1;
+ 
+-			device->postEventFromUser(event);
++			status = device->postEventFromUser(event) ? 1 : 0;
```
Comment (Zeno-) [inline build/android/patches/irrlicht-back_button.patch]: Why the ternary operator/branch? device->postEventFromUser(event) returns a type (bool) compatible with status anyway...

Comment (est31) [inline build/android/patches/irrlicht-back_button.patch]: Does the C++ standard define how bool gets stored? status **must** be 0 if the return value is false, and **must** be 1 if the return value is true, no other values are allowed.

Are you proposing a cast? Or does c++ cast it automatically

Comment (Zeno-) [inline build/android/patches/irrlicht-back_button.patch]: The Standard defines the value of false to be 0 and true to be 1. It should work (without warnings) without a cast IIRC

**Comment by est31** [comment]
Closing in favor of #4086 , It has applied all changes and it seems the onBackPressed override was required after all.

I've looked into it and believe that onBackPressed is required because of a bug in minetest, and its better to fix the bug, but that's better done in a separate PR and merged after the release.

**Comment by MoNTE48** [comment]
@est31 onBackPressed is not error in minetest or jni. This is default android action http://stackoverflow.com/questions/4779954/disable-back-button-in-android

**Pull request closed without merging**